src/plugin/systems/collider.rs

```diff
@@ -143,75 +176,87 @@ pub fn apply_collider_user_changes(
         }
     }
 
-    for (handle, active_events) in changed_active_events.iter() {
+    for (rapier_entity, handle, active_events) in changed_active_events.iter() {
+        let context = context.context(*rapier_entity.rapier_context_link);
```

Comment:
Could this API be changed to not have to dereference the rapier context link every time?

Reply:
What's wrong with dereferencing ? I'm assuming change detection can get in the way ; but I didn't notice much performance difference.

I considered caching the rapiercontext, or preordering entities before iterating on them, but I feared the borrow checker would yell at me and as I didn't measure a significant performance I didn't dig into it. Should I ? Or consider it a follow up ?

Reply:
or you mean dereference the link 💡 ; sure :)

Reply:
to be noted the dereference is only an `Entity`, leading to a copy of said entity, which is quite idiomatic in bevy. As I don't have a strong opinion about this and consider code without the dereference a liiittle easier on the eyes, I went ahead on this change.

:warning: Unfortunately I went the exra mile and changed the signature to return a `Mut<RapierContext>` , which is more powerful if we don't want to trigger change mutability... so I did trade the `*` for a `to_inner` (could be a `&mut *`)...

Reply:
I like that change even more =)